pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractBaseDispatcher.java

```diff
@@ -392,6 +392,10 @@ public static void checkAndApplyReachedEndOfTopicOrTopicMigration(PersistentTopi
         }
     }
 
+    protected EntryBatchIndexesAcks getEntryBatchIndexesAcks(int entries) {
```

Comment:
Giving it a `@Nullable` annotation to returned value is better.

Reply:
+1

Reply:
Maybe we could create an class `DisableEntryBatchIndexesAcks` and return it if `serviceConfig.isAcknowledgmentAtBatchIndexLevelEnabled()` is false  to avoid NPE

Reply:
Agree. We can override the method and return 0.

Reply:
Without a new interface or a base class, we are not able to do so. I don't want to introduce a new interface here.